File: 03-FRONTEND/src/services/LocationService.js
import api from './api';

const LocationService = {
  getPage: async (params = {}) => {
    const response = await api.get('/locations', { params });
    return response.data;
  },

  getAll: async () => {
    const response = await api.get('/locations/all');
    return response.data;
  },

  getById: async (id) => {
    const response = await api.get(`/locations/${id}`);
    return response.data;
  },

  create: async (data) => {
    const response = await api.post('/locations', data);
    return response.data;
  },

  update: async (id, data) => {
    const response = await api.put(`/locations/${id}`, data);
    return response.data;
  },

  delete: async (id) => {
    const response = await api.delete(`/locations/${id}`);
    return response.data;
  },

  getProductLocations: async (productId) => {
    const response = await api.get(`/locations/product/${productId}`);
    return response.data;
  },

  updateProductLocationStock: async (productId, locationId, stock) => {
    const response = await api.post(`/locations/product/${productId}/stock`, null, {
      params: { locationId, stock }
    });
    return response.data;
  },

  transferStock: async (productId, fromLocationId, toLocationId, quantity) => {
    const response = await api.post(`/locations/product/${productId}/transfer`, null, {
      params: { fromLocationId, toLocationId, quantity }
    });
    return response.data;
  },
};

export default LocationService;

File: 02-BACKEND/supermarket-system-api/src/main/java/com/supermarket/product/service/LocationService.java
package com.supermarket.product.service;

import java.math.BigDecimal;
import java.util.List;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import com.supermarket.product.dto.LocationRequestDTO;
import com.supermarket.product.dto.LocationResponseDTO;
import com.supermarket.product.dto.ProductLocationResponseDTO;

public interface LocationService {
	List<LocationResponseDTO> findAll();
	Page<LocationResponseDTO> findPage(String search, Pageable pageable);
	LocationResponseDTO findById(Long id);
	LocationResponseDTO create(LocationRequestDTO request);
	LocationResponseDTO update(Long id, LocationRequestDTO request);
	void deleteById(Long id);
	List<ProductLocationResponseDTO> getProductLocations(Long productId);
	void updateProductLocationStock(Long productId, Long locationId, BigDecimal stock);
	void transferStock(Long productId, Long fromLocationId, Long toLocationId, BigDecimal quantity);
	List<ProductLocationResponseDTO> getProductsByLocation(Long locationId);
	void removeProductLocation(Long productId, Long locationId);
}
